src/apim_operations.ts

```diff
@@ -58,7 +62,10 @@ function getToken(
 }
 
 export async function loginToApim(
-  tokenCreds?: ITokenAndCredentials
+  tokenCreds?: ITokenAndCredentials,
+  servicePrincipalClientId?: string,
```

Comment:
since we need all three or neither, I would ask for an optional object with all of them

```suggestion
  servicePrincipalCredentials?: {
    clientId: string;
    secret: string;
    tenantId: string;
  },
```